client/src/storage/storage.ts

```diff
@@ -0,0 +1,24 @@
+import {IGenomeData} from "../genetics/genome"
+import {Hexalot} from "../island/hexalot"
+import {IslandPattern} from "../island/island"
+import {Journey} from "../island/journey"
+
+export interface IStorage {
+
+    getIsland(islandName: string): Promise<IslandPattern>
+
+    claimHexalot(hexalot: Hexalot): Promise<IslandPattern | undefined>
```

Comment:
On my side, this method expects a genomeData param. That's what you're claiming the hexalot for after all

Reply:
Also, I need an islandName param.